homeassistant/helpers/script.py

```diff
@@ -909,6 +913,33 @@ async def async_run_sequence(iteration, extra_msg=""):
                     _LOGGER.warning("Error in 'while' evaluation:\n%s", ex)
                     break
 
+                if iteration > 1:
+                    # If the user creates an script with a tight loop,
+                    # yield to the event loop so the system stays
+                    # responsive while all the cpu time is consumed.
+                    await asyncio.sleep(0)
+                    if iteration >= REPEAT_WARN_ITERATIONS:
+                        if not warned_too_many_loops:
+                            warned_too_many_loops = True
+                            _LOGGER.warning(
+                                "While condition %s in script `%s`"
+                                " is looping more than %s times",
+                                repeat[CONF_WHILE],
+                                self._script.name,
+                                REPEAT_WARN_ITERATIONS,
+                            )
```

Comment:
Should we register an issue here?

Reply:
I think it depends on if we want to backport this to 2024.4.x or not. Typically we wouldn't backport a breaking change, but this one seems worth fixing it since the impact of a bad loop is rather high.

Reply:
I added the milestone. I didn't do the issue since it already shows up in the trace and we don't have a precident for creating issues from `script.py`

<img width="897" alt="Screenshot 2024-04-07 at 10 24 00 AM" src="https://github.com/home-assistant/core/assets/663432/ee0f1cbc-39e3-4311-8f80-8349c83fb6ce">